ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingHooks.kt

```diff
@@ -0,0 +1,21 @@
+package io.ktor.server.routing
+
+import io.ktor.server.application.*
+
+/**
+ * Hook that will be triggered when a call was routed and before any other handlers.
+ */
+public object CallRouted : Hook<(RoutingApplicationCall) -> Unit> {
```

Comment:
consider making all hooks internal and move to the plugin scope for a while. Let's have separate discussion about hooks and naming